sklearn/metrics/_ranking.py

```diff
@@ -525,14 +526,21 @@ def _binary_clf_curve(y_true, y_score, pos_label=None, sample_weight=None):
         sample_weight = column_or_1d(sample_weight)
 
     # ensure binary classification if pos_label is not specified
+    # _determine_key_type(classes) == 'str' is required to avoid
+    # triggering a FutureWarning by calling np.array_equal(a, b)
+    # where a has string values and b has integer values.
```

Comment:
perhaps ``when elements in the two arrays are not comparable``? 
you'll get the same warning for ``np.array_equal(["a", "b"], [1.0, 2.0])``